homeassistant/components/mqtt/discovery.py

```diff
@@ -111,7 +141,7 @@ def async_log_discovery_origin_info(
 
 @callback
 def _replace_abbreviations(
-    payload: Any | dict[str, Any],
+    payload: dict[str, Any] | str,
```

Comment:
Is this change really correct? How can we know `payload` is a dict of strings or a string, this function is called before any validation, right?